### DISCUSSION TITLE
【提案】允许在同一作用域内重新声明变量

### DISCUSSION THREAD

**Discussion by tbontb-iaq** [JSCIG/es-discuss#67] [Ideas]
就像 Rust 一样：

```javascript
let n = 1
const obj = {}

// 重新声明
let n = 10
const obj = { key: 'value' }
```

**Comment by ljharb** [comment]
Why? That's generally considered one of the bad parts of `var`, and something that `let` and `const` fixed.

**Reply by tbontb-iaq** [reply]
I don't think this is a bad part. It is useful to allow re-declare variables like Rust. It can reduce the number of times of developers' thinking variable names. The problem of `var` should be that there is no block -level domain and variable hoisting.

**Reply by ljharb** [reply]
Can you share a non-contrived example where both variables need to be in the same scope, and where it would be difficult to select two different names?
